.sharingio/init

```diff
@@ -0,0 +1,52 @@
+#!/bin/env bash
+set -x
+
+# upgrade go to 1.19.1
+go version | grep 1.19.1 || curl -L https://dl.google.com/go/go1.19.1.linux-amd64.tar.gz | sudo tar --directory /usr/local --extract --ungzip
+# shfmt needed for make
+which shfmt || sudo apt-get install shfmt
+# TODO: Make still failing, possible dependencies still missing.
+
+# install coder binary until we can build from src
+which coder || (
+	curl -OL https://github.com/coder/coder/releases/download/v0.9.1/coder_0.9.1_linux_amd64.deb
+	sudo dpkg -i coder_0.9.1_linux_amd64.deb
+	# Add completion
+	echo '. <(coder completion bash)' >>~/.bashrc
+)
+
+# Deploying coder (from helm for now)
+kubectl create namespace coder
+# ensure ingress works / certs secrets get copied
+kubectl label ns coder cert-manager-tls=sync
+# needs a postgres db
+helm repo add bitnami https://charts.bitnami.com/bitnami
+helm install postgres bitnami/postgresql \
+	--namespace coder \
+	--set auth.username=coder \
+	--set auth.password=coder \
+	--set auth.database=coder \
+	--set persistence.size=10Gi
+# deploy via helm for now
+envsubst <.sharingio/values.template.yaml >.sharingio/values.yaml
+helm install coder ./helm/ \
+	--namespace coder \
+	--values .sharingio/values.yaml
+# setup ingress
+envsubst <.sharingio/ingress.template.yaml >.sharingio/ingress.yaml
+kubectl apply -f .sharingio/ingress.yaml
+# Wait for coder to deploy
+kubectl rollout status deployment coder -n coder
+kubectl wait -n coder --for=condition=ready pod -l app.kubernetes.io/name=coder
+
+# create the initial user
+# populate ii or pair as an admin user without logging in
+CODER_EMAIL=ii@ii.coop
+CODER_PASSWORD=ii
+CODER_USERNAME=ii
+CODER_URL=https://coder.${SHARINGIO_PAIR_BASE_DNS_NAME}
+# export vars to we can emulate a tty with a short expect script
+export CODER_EMAIL CODER_PASSWORD CODER_USERNAME
+coder login $CODER_URL -u $CODER_USERNAME -p $CODER_PASSWORD -e $CODER_EMAIL
+
+# TODO : upload / update the kubernetes template
```

Comment:
Currently unable to push / create or update templates due to 403 error:
```
+ kubectl rollout status deployment coder -n coder
Waiting for deployment "coder" rollout to finish: 0 of 1 updated replicas are available...
deployment "coder" successfully rolled out
+ kubectl wait -n coder --for=condition=ready pod -l app.kubernetes.io/name=coder
pod/coder-865b875674-n9sh5 condition met
+ CODER_EMAIL=ii@ii.coop
+ CODER_PASSWORD=ii
+ CODER_USERNAME=ii
+ CODER_URL=https://coder.pair.pair.sharing.io
+ export CODER_EMAIL CODER_PASSWORD CODER_USERNAME
+ coder login https://coder.pair.pair.sharing.io -u ii -p ii -e ii@ii.coop
> Your Coder deployment hasn't been set up!
                                                            
  Welcome to Coder, ii! You're authenticated.               
                                                            
  Get started by creating a template:  coder templates init 
ii@pair:~/sharingio/coder$ coder templates list
NAME        LAST UPDATED        USED BY              
kubernetes  September 28, 2022  - active developers  
ii@pair:~/sharingio/coder$ coder templates push kubernetes -d examples/templates/kubernetes -v -y
⧗  Queued 
.                                           
logs: unexpected status code 403:           
Run 'coder templates push --help' for usage.
```

Reply:
Both `make` and `go run` still seem to be missing something:
```
ii@pair:~/sharingio/coder$ make build/coder-slim_linux_amd64
ii@pair:~/sharingio/coder$ go run ./cmd/coder
# sort
/usr/local/go/src/sort/zsortfunc.go:10:6: insertionSort_func redeclared in this block
        /usr/local/go/src/sort/zfuncversion.go:10:6: other declaration of insertionSort_func
/usr/local/go/src/sort/zsortfunc.go:20:6: siftDown_func redeclared in this block
        /usr/local/go/src/sort/zfuncversion.go:19:6: other declaration of siftDown_func
/usr/local/go/src/sort/zsortfunc.go:38:6: heapSort_func redeclared in this block
        /usr/local/go/src/sort/zfuncversion.go:38:6: other declaration of heapSort_func
/usr/local/go/src/sort/zsortfunc.go:329:6: swapRange_func redeclared in this block
        /usr/local/go/src/sort/zfuncversion.go:65:6: other declaration of swapRange_func
/usr/local/go/src/sort/zsortfunc.go:335:6: stable_func redeclared in this block
        /usr/local/go/src/sort/zfuncversion.go:163:6: other declaration of stable_func
/usr/local/go/src/sort/zsortfunc.go:378:6: symMerge_func redeclared in this block
        /usr/local/go/src/sort/zfuncversion.go:187:6: other declaration of symMerge_func
/usr/local/go/src/sort/zsortfunc.go:464:6: rotate_func redeclared in this block
        /usr/local/go/src/sort/zfuncversion.go:252:6: other declaration of rotate_func
```

Reply:
The go issues were resolved by `rm -rf /usr/local/go` before upgrading golang.

Reply:
Go 1.19.1 was directly installed over top of Go 1.18.3 and caused this issue.

A quick fix was
```
sudo rm -rf /usr/local/go/
. /usr/lib/sharingio/environment/helper.sh
curl -L https://dl.google.com/go/go1.19.1.linux-${ARCH_TYPE_2}.tar.gz     | tar --directory /usr/local --extract --ungzip
```

sharingio/environment was updated to include Go 1.19.1

The line causing the issue was
https://github.com/sharingio/coder/blob/1a899ab/.sharing.io/init#L5